src/layout.rs

```diff
@@ -59,7 +59,10 @@ use State::*;
 /// key i=2, j=3 on the layer 1.
 pub type Layers<T = core::convert::Infallible> = &'static [&'static [&'static [Action<T>]]];
 
-type Stack = ArrayDeque<[Stacked; 16], arraydeque::behavior::Wrapping>;
+/// The current event stack.
+///
+/// Events can be retrieved by iterating over this struct and calling [Stacked::event].
+pub type Stack = ArrayDeque<[Stacked; 16], arraydeque::behavior::Wrapping>;
```

Comment:
I think it would be better to avoid exposing this type directly, but only exposing some new type wrapping an iterator, allowing to change the Stack type without a breaking change.

```rust
struct StackedIter<'a>(arraydeck::Iter<'a, Stacked>);
impl<'a> Iterator for StackedIter {
    type Item = &'a Stacked;
    fn next(&mut self) -> Option<Self::Item> {
        self.0.next()
    }
    fn size_hint(&self) -> (usize, Option<usize>) {
        self.0.size_hint()
    }
}
```

Reply:
Use this. Thanks!